### PR TITLE
Load real data to the availability page

### PR DESCRIPTION
**Pull request by olena-kozak**
Case if user don't have own slots:
image
Case if user have own slots, but don't have slots connected with teams:
image
Case when IsVisible = false: 
image
Response when getting teams' slots by user:
image
Response when getting all available user slots:
image

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 1]
```diff
@@ -1,29 +1,59 @@
...
+            var teamsWithSlots = await _context.Teams
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Members)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Location)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Author)
+                .Where(x => x.AvailabilitySlots.Any(x => x.Type == SlotType.Team))
+                .Where(x => x.AvailabilitySlots.Any(x => x.Members.Any(x => x.UserId == id)))
+                .ToListAsync();
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: This is not a good idea to load all data and only after that map it to a DTO
It would be better to do a projection here and load only necessary fields

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
+            var userSlots = await _context.Users
+                 .Where(x => x.Id == id)
+                 .Include(x => x.CreatedSlots)
+                    .ThenInclude(x => x.Members)
+                    .ThenInclude(x => x.User)
+                 .Include(x => x.CreatedSlots)
+                    .ThenInclude(x => x.Location)
+                 .Where(x => x.CreatedSlots.Any(x => x.Type == SlotType.Personal))
+                 .SelectMany(x => x.CreatedSlots)
+                 .ToListAsync();
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: See the previous comment

File: backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs [hunk 3]
```diff
@@ -2,6 +2,6 @@ namespace EasyMeets.Core.Common.Enums;
 
 public enum SlotType
 {
-    Personal,
-    Team
+    Personal = 1,
+    Team = 2
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: Why did you change enum numbering?

File: frontend/src/app/core/models/availiability-slot.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface AvailabilitySlot {
+    name: string;
+    author: User;
+    size: number;
+    link: string;
+    isEnabled: boolean
+    isVisible: boolean
+    location: Location
+    members: Array<User>
```
Comment (tatianahutii) [inline frontend/src/app/core/models/availiability-slot.ts]:
```suggestion
    members: User[]
```

File: frontend/src/app/core/models/team-with-slot.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { AvailabilitySlot } from './availiability-slot';
+
+export interface TeamWithSlots {
+    name: string
+    pageLink: string
+    availabilitySlots: Array<AvailabilitySlot>
```
Comment (tatianahutii) [inline frontend/src/app/core/models/team-with-slot.ts]:
```suggestion
    availabilitySlots: AvailabilitySlot[]
```

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts [hunk 6]
```diff
@@ -1,8 +1,57 @@
...
     templateUrl: './availability-page.component.html',
     styleUrls: ['./availability-page.component.sass'],
 })
-export class AvailabilityPageComponent {}
+export class AvailabilityPageComponent {
+    public teams: Array<TeamWithSlots>;
+
+    public currentUser: User;
+
+    public userSlots: Array<AvailabilitySlot>;
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.ts]:
```suggestion
    public userSlots: AvailabilitySlot[];
```

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts [hunk 7]
```diff
@@ -1,8 +1,57 @@
...
+            .subscribe(
+                (resp) => {
+                    this.userSlots = resp;
+                },
+            );
+    }
+
+    public getCurrentUser() {
+        this.userService
+            .getCurrentUserById(4)
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: It would be better to not hard code the number
At least we need to move it to a separate variable

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs (hunk 3 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: I changed enum numbering because slot type in table is saved as value 1 and 2 (maybe this error was made during seeding). 
Because of that, there was error during loading data from database. To remove this error and convenient enum mapping, I set the value as it is in db. 
So, I made it so that the enum value matched the values in the database and avoided adding (+1).

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: It would be better to fix seed and db values

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 8]
```diff
@@ -21,13 +23,52 @@ public async Task<ICollection<AvailabilitySlotsGroupByTeamsDto>> GetAllAvailabil
...
+                                             Id = m.User.Id,
+                                             UserName = m.User.Name,
+                                             Email = m.User.Email,
+                                             Image = m.User.ImagePath,
+                                             Phone = m.User.PhoneNumber
+                                         })
+                                        .ToList()
+                                })
+                            .ToList()
+                    })
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Are you really need all this field?
You only need to load data that you will use on your page

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 8 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Yes, I need these data, I can remove user email and phone number of user (maybe also link in availability slot for now, but it will be used later, so I decided to leave it).
But other fields are displaying on availability page.
image

File: backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs (hunk 3 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: So, enum value in db should start with 0?

Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: Can I fix this bug in separate branch, I think it doesn't refer to this task and so that every will know what happened and load new db after fixing seed?

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: Yes, sure
Could you please create a card for this?

Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.Common/Enums/SlotType.cs]: Of course. 
So, I commit with wrong (Personal=1) data, so that my page will receive data. And in another brunch I will fix seed and change enum numbering, is it right?

**Review by tatianahutii — CHANGES_REQUESTED** [review]
Could you please fix warnings?
image

**Review by tatianahutii — CHANGES_REQUESTED** [review]
Still see a few git warnings

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 9]
```diff
@@ -1,23 +1,119 @@
...
+                            },
+                        Members = y.Members
+                            .Select(m =>
+                                new UserDto
+                                {
+                                    UserName = m.User.Name,
+                                    Image = m.User.ImagePath,
+                                })
+                            .ToList()
+                    })
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Could you simplify this to one layer DTO?

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 10]
```diff
@@ -1,23 +1,119 @@
...
+                                        Name = y.Location.Name,
+                                    },
+                                    Members = y.Members
+                                        .Select(m =>
+                                         new UserDto
+                                         {
+                                             UserName = m.User.Name,
+                                             Image = m.User.ImagePath,
+                                         })
+                                        .ToList()
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: You can create simple DTO and use mapper here

**Review by tatianahutii — CHANGES_REQUESTED** [review]
Please use icon instead of png
image

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 9 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: Yes.
And leave projection here or use mapper?

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 10 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: I simplified dto and used mapper in this method

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 9 above)
Comment (olena-kozak) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]: I simplified dto and use projection in this method, so that unused data will not be load in this method

**Review by tatianahutii — APPROVED** [review]
LGTM

**Review by tatianahutii — CHANGES_REQUESTED** [review]
Need recreate megration

**Review by tatianahutii — APPROVED** [review]
LGTM
